src/snowflake/snowpark/dataframe.py

```diff
@@ -3176,7 +3172,9 @@ def with_column_renamed(self, existing: ColumnOrName, new: str) -> "DataFrame":
                     f"Unable to rename column {existing} because it doesn't exist."
                 )
         else:
-            raise TypeError("'exisitng' must be a column name or Column object.")
+            raise TypeError(
```

Comment:
This is an API call. Should we have a negative test?

Reply:
I don't think we need to test for negative cases of input that are indicated by the type hints already, but I'm open to discussion :)

Reply:
Oh I'm not aware of that lol. Github just shows this line to me. It does not hurt to test but yeah it's also fine to leave it as is. No strong opinion then.